### DISCUSSION TITLE
"translate" sql query to cli

### DISCUSSION THREAD

**Discussion by crasty01** [supabase/supabase#2409] [Questions] [answered]
I have made this sql query:

```sql
select count("id"), users."display-name" from public.chat
LEFT JOIN users ON chat."user-id" = users."user-id"
where chat.datetime >= current_timestamp - interval '7 days'
group by users."display-name"
order by count("id") desc
limit 10
```
but don't know, how to "translate" it to javascript supabase cli. Is there a way? Or are there any other options?

Thanks in advance.

**Comment by crasty01** [comment, marked as answer]
Already figured it out - views.

**Reply by djeglin** [reply]
Any chance you could share your solution? I'm currently trying to translate a SQL query doing a simple filter based on timestamp to the JS library and its not filtering the way it should do

**Reply by soedirgo** [reply]
@djeglin you can create a view like this:
```sql
create view my_view as
  select count("id"), users."display-name" from public.chat
  LEFT JOIN users ON chat."user-id" = users."user-id"
  where chat.datetime >= current_timestamp - interval '7 days'
  group by users."display-name"
  order by count("id") desc
  limit 10;
```

But if you just want a simple filter you might not need this - if you'd like we can discuss your specific case on a new Discussion.
